index.js

```diff
@@ -64,8 +75,72 @@ module.exports = (bookshelf, settings) => {
     destroy: function (options) {
       options = options || {}
       if (this.softDelete === true && options.hardDelete !== true) {
-        options.patch = true
-        return this.save({ [settings.field]: new Date() }, options)
+        // Add default values to options
+        options = merge(options, {
+          method: 'update',
+          patch: true,
+          softDelete: true
+        })
+
+        // Attributes to be passed to events
+        let attrs = { [settings.field]: new Date() }
+
+        return Promise.resolve()
+        .then(() => {
+          // Don't need to hooks if there's no events registered
+          if (!settings.events) return
+
+          let events = []
+
+          // Emulate all pre update events
+          if (settings.events.destroying) {
+            events.push(this.triggerThen('destroying', this, options))
+          }
+
+          if (settings.events.saving) {
+            events.push(this.triggerThen('saving', this, attrs, options))
+          }
+
+          if (settings.events.updating) {
+            events.push(this.triggerThen('updating', this, attrs, options))
+          }
+
+          // Resolve all promises in parallel like bookshelf does
+          return Promise.all(events)
+        })
+        .then(() => {
+          let knex = options.transacting || bookshelf.knex
+
+          return knex(this.tableName)
+          .update(attrs, this.idAttribute)
```

Comment:
knex doesn't accept any bookshelf options, do you think that we're missing something?